plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/preferences/SSLCertificatesPreference.java

```diff
@@ -80,128 +111,114 @@ public void setAllowedByCertificate(X509Certificate certificate, boolean result)
 	 * 
 	 * @return the single list instance, which is never replaced, only its content is changed
 	 */
-	List<Item> getSavedItems() {
-		if(savedItems == null) {
-			List<Item> savedItems = new ArrayList<>();
-			String value = getPreferenceStore().getString(ALLOWED_CERTIFICATES);
-			if(value != null && value.length() > 0) {
-				String[] tokens = value.split(SEPARATOR);
-				for (int i = 0; i + 4 <= tokens.length;) {
-					boolean checked = "true".equals(tokens[i++]);
-					String issuedBy = tokens[i++];
-					String validity = tokens[i++];
-					String fingerprint = tokens[i++];
-					if(isValid(validity)) {
-						savedItems.add(new Item(checked, issuedBy, validity, fingerprint));
-					}
+	List<HostCertificate> getSavedCertificates() {
+		if (savedItems == null) {
+			this.savedItems = loadSavedCertificates();
+		}
+		return savedItems;
+	}
+
+	private List<HostCertificate> loadSavedCertificates() {
+		List<HostCertificate> savedItems = new ArrayList<>();
+		String value = getPreferenceStore().getString(ALLOWED_CERTIFICATES_NEW);
+		if (!StringUtils.isBlank(value)) {
+			String[] 	tokens = value.split(SEPARATOR);
+			for (int i = 0; i + 4 <= tokens.length;) {
```

Comment:
Plus I noticed that there is a missing separator between the end of the certificate and the start of the following. So I stored 2 certificates but when I restarted the list was empty because we got only 3 tokens thus the while is not executed